File: src/test/java/MockRange.java
import shootemup.geometry.IRange;
import java.util.Set;
import java.util.HashSet;


public class MockRange implements IRange {

    public MockRange(boolean returnValue) {
	this.returnValue = returnValue;
	this.valuesUsed = new HashSet<>();
    }

    public MockRange() {
	this(false);
    }

    @Override
    public boolean contains(int value) {
	valuesUsed.add(value);
	return returnValue;
    }

    public boolean calledWith(int value) {
	return valuesUsed.contains(value);
    }

    private final boolean returnValue;
    private final Set<Integer> valuesUsed;
}

File: src/test/kotlin/mocks/MockRange.kt
package mocks

import shootemup.IRange

class MockRange(private val returnValue: Boolean = false): IRange {

  private val valuesUsed: MutableSet<Int>

  init{
    this.valuesUsed = hashSetOf()
  }

  override fun contains(value: Int): Boolean {
    valuesUsed.add(value)
    return returnValue
  }

  fun calledWith(value: Int) = valuesUsed.contains(value)
}
